### PR TITLE
Break serviceaffinity Filter plugins dependency on predicates package

### PR DESCRIPTION
**Pull request by haosdent**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Scheduler predicates and their error types are deprecated. The Scheduler moved to the plugins framework. At the same time, the serviceaffinity filter plugin does not require any of the logic in predicates.

**Which issue(s) this PR fixes**:

Refs #86712

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by haosdent** [comment]
cc @ahg-g

**Comment by haosdent** [comment]
thx @tanjunchen !

File: pkg/scheduler/framework/plugins/serviceaffinity/utils_test.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+			},
+		},
+	}
+	fmt.Println(FindLabelsInSet([]string{"label1", "label2", "label3"}, nsPods[0].ObjectMeta.Labels)["label3"])
+	AddUnsetLabelsToMap(labelSubset, []string{"label1", "label2", "label3"}, nsPods[0].ObjectMeta.Labels)
+	fmt.Println(labelSubset)
+
+	for _, pod := range FilterPodsByNamespace(nsPods, "ns1") {
+		fmt.Print(pod.Name, ",")
+	}
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/serviceaffinity/utils_test.go]: This is not testing anything, please either add proper testing (i.e., the output should be compared with something), or just remove it complete if we think service_affinity_test.go has enough coverage.

**Comment by ahg-g** [comment]
haosdent@ Are you still working on this? are you able to finish it?

**Comment by haosdent** [comment]
@ahg-g sry miss the comment, let me update

**Comment by haosdent** [comment]
@ahg-g updated, please help to review again when you are available. Sorry that miss comments previously.

File: pkg/scheduler/framework/plugins/serviceaffinity/service_affinity.go [hunk 2]
```diff
@@ -390,3 +390,47 @@ func (pl *ServiceAffinity) updateNodeScoresForLabel(sharedLister schedulerlister
...
+		if nsPod.Namespace == ns {
+			filtered = append(filtered, nsPod)
+		}
+	}
+	return filtered
+}
+
+// FindLabelsInSet gets as many key/value pairs as possible out of a label set.
+func FindLabelsInSet(labelsToKeep []string, selector labels.Set) map[string]string {
+	aL := make(map[string]string)
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/serviceaffinity/service_affinity.go]: please make all those functions local:

s/AddUnsetLabelsToMap/addUnsetLabelsToMap
s/CreateSelectorFromLabels/createSelectorFromLabels
s/FilterPodsByNamespace/filterPodsByNamespace
s/FindLabelsInSet/findLabelsInSet

**Comment by haosdent** [comment]
fixed, thx @ahg-g . Could you help to review again? Many thx!

**Pull request merged**